src/Simplex/Chat/Types.hs

```diff
@@ -1637,6 +1637,15 @@ data CommandData = CommandData
   }
   deriving (Show)
 
+data ChatTag = ChatTag
+  { chatTagId :: Int64,
+    chatTagText :: Text,
+    chatTagEmoji :: Text,
+    contactId :: Maybe ContactId,
+    groupId :: Maybe GroupId
+  }
```

Comment:
I don't seem to understand the model / how it will work with ui - tags are not on chat type, but also tag has single chat and not list. So:
1. when / if tags are to be rendered for chat, all tags have to be scanned for including chat
2. when displaying tags, duplicate tags have to be filtered out in ui, but when tapping on them, ui has to find all tags with same text and emoji to filter corresponding conversations

Database has many-to-many, why at least ChatTag doesn't have list of chats instead of single chat? At least problem 2 wouldn't require additional logic in ui - tags (text/emoji) will be unique and have list of chats to filter. Since it's already in schema, why are tags loaded in such a strange way?
Problem 1 would still require searching in all tags but it's also present in current design and should be ok with small number of chats

Reply:
yes, it doesn't make any sense. contactId and groupId are simply not needed here.